src/cluster.c

```diff
@@ -4440,6 +4440,110 @@ void restoreCommand(client *c) {
     server.dirty++;
 }
 
+void prepareRollback(struct redisCommand *cmd, client *c) {
```

Comment:
Two reasons:
1. I didn't know where else to put it (so I am happy to move it)
2. The current implementation is an automatic dump/restore
3. Most of the guts of dump/restore are defined in cluster.c, implemented earlier in the file

Where should I move it?